plugins/distance_table.hpp

```diff
@@ -70,24 +70,34 @@ template <class DataFacadeT> class DistanceTablePlugin final : public BasePlugin
     int HandleRequest(const RouteParameters &route_parameters,
                       osrm::json::Object &json_result) override final
     {
-        if (!check_all_coordinates(route_parameters.coordinates))
+        const bool useSameTgtSrc = route_parameters.coordinates.size() ? true : false;
+        if ((useSameTgtSrc && route_parameters.destinations.size()) || (useSameTgtSrc && route_parameters.sources.size()))
+        {
+            return 400;
+        }
+
+        if ((useSameTgtSrc && !check_all_coordinates(route_parameters.coordinates)) ||
+            (!useSameTgtSrc && !check_all_coordinates(route_parameters.destinations, 2) && !check_all_coordinates(route_parameters.sources, 1)))
         {
             return 400;
         }
 
         const auto &input_bearings = route_parameters.bearings;
-        if (input_bearings.size() > 0 && route_parameters.coordinates.size() != input_bearings.size())
+        unsigned nb_coordinates = useSameTgtSrc ? route_parameters.coordinates.size() : (route_parameters.destinations.size() + route_parameters.sources.size());
+        if (input_bearings.size() > 0 && nb_coordinates != input_bearings.size())
         {
             json_result.values["status"] = "Number of bearings does not match number of coordinates .";
             return 400;
         }
 
         const bool checksum_OK = (route_parameters.check_sum == facade->GetCheckSum());
-        unsigned max_locations =
+        unsigned max_locations = useSameTgtSrc ?
```

Comment:
I guess it would be cleaner to just return `400` for tables that exceed a size of `max_locations_distance_table*max_locations_distance_table` so this should be more like:

```c++
if (number_of_sources*number_of_destinations > max_locations_distance_table*max_locations_distance_table)
{
    json_result.values["status_message"] = std::string("Table exceeds maximum number of entries of ") + std::to_string(max_locations_distance_table*max_locations_distance_table);
    return 400;
}
```

Where `auto number_of_sources = route_parameters.coordinates.size() + route_parameters.sources.size()` and `auto number_of_destinations = route_parameters.coordinates.size() + route_parameters.destinations.size();`